tsfresh/feature_extraction/extraction.py

```diff
@@ -208,28 +207,31 @@ def _do_extraction(df, column_id, column_value, column_kind,
     :param disable_progressbar: Do not show a progressbar while doing the calculation.
     :type disable_progressbar: bool
 
-    :param distributor: Advanced parameter: set this to an instance name that you want to use as a
-             distributor. See the utilities/distribution.py for more information. Leave to None, if you want
-             TSFresh to choose the best distributor.
-    :type distributor: Distributor
+    :param distributor: Advanced parameter:  See the utilities/distribution.py for more information.
+                         Leave to None, if you want TSFresh to choose the best distributor.
+    :type distributor: DistributorBaseClass
 
     :return: the extracted features
     :rtype: pd.DataFrame
     """
     data_in_chunks = [x + (y,) for x, y in df.groupby([column_id, column_kind])[column_value]]
 
     if distributor is None:
+
         if n_jobs == 0:
-            distributor_class = MapDistributor
+            distributor = MapDistributor(disable_progressbar=disable_progressbar,
```

Comment:
you could summarize the arguments into a dict and fill it in in both cases, to reduce code duplication.

Reply:
I think it is quite easy to read in its current state...